src/main/java/kr/sesac/aoao/server/dino/service/DinoServiceImpl.java

```diff
@@ -43,14 +54,83 @@ public GetUserDinoResponse getDinoInfo(Long userId) {
 			.orElseThrow(() -> new ApplicationException(UserErrorCode.NOT_FOUND_USER));;
 		DinoEntity dino = dinoRepository.findByUser(user)
 			.orElseThrow(() -> new ApplicationException(DinoErrorCode.NO_DINO));
-		return new GetUserDinoResponse(
-			dino.getUser().getId(),
-			dino.getName(),
-			dino.getColor(),
-			dino.getExp(),
-			dino.getDino().getLv(),
-			dino.getPoint()
-		);
+		return result(dino);
 	}
 
+	/**
+	 * 다이노 이름 변경
+	 * @since 2024.01.22
+	 * @return GetUserDinoResponse
+	 * @author 김은서
+	 */
+	@Override
+	public GetUserDinoResponse renameDino(Long dinoId, String name) {
+		DinoEntity dino = dinoRepository.findById(dinoId)
+			.orElseThrow(() -> new ApplicationException(DinoErrorCode.NO_DINO));
+		dino.changeName(name);
+		dinoRepository.save(dino);
+		return result(dino);
+	}
```

Comment:
```suggestion
	@Override
	public GetUserDinoResponse renameDino(Long dinoId, String name) {
		DinoEntity dino = dinoRepository.findById(dinoId)
			.orElseThrow(() -> new ApplicationException(DinoErrorCode.NO_DINO));
		dino.changeName(name);
		return result(dino);
	}
```

JPA 는 변경 감지 속성이 있기 때문에 repository 를 이용하여 가져온 객체에 한해(영속성 객체) 특정 필드가 수정되면 트랜잭션이 종료될때 변경된 값들을 수정하는 쿼리문을 짜주는 속성이 있습니다!! 따라서 save 문은 다시 호출하지 않아도 됩니다 ㅎㅎ